irods/session.py

```diff
@@ -51,6 +89,8 @@ def __init__(self, configure=True, **kwargs):
         self.resources = ResourceManager(self)
         self.zones = ZoneManager(self)
 
+        _weakly_reference(self)
```

Comment:
create issue for the lines above, to the effect the manager subobjects could refer to their parent session with a weak reference to help break the reference cycles known to exist.

Reply:
Tested this approach, in general a failing one since objects (iRODSCollections, etc) gotten from sessions are known to refer to the parent session and possibly outlive them, leaving those weak references null.